javalib/src/main/scala/java/util/AbstractSet.scala

```diff
@@ -0,0 +1,48 @@
+package java.util
+
+import scala.annotation.tailrec
+
+abstract class AbstractSet[E]() extends java.util.AbstractCollection[E] with java.util.Set[E] {
+
+  override def equals(o: Any): Boolean = {
+    def _equals =
+      o match {
+        case s: java.util.Collection[E] =>
+          if (s.size == this.size) containsAll(s)
+          else false
+        case _ => false
+      }
+
+    o match {
+      case ar: AnyRef =>
```

Comment:
Just cast `o.asInstanceOf[AnyRef]`. This always succeeds. Your `case _ =>` is dead code.